### ISSUE TITLE
Error "Not an editor command: commentstring" after BufRead on jsonc buffer

### ISSUE THREAD

**Issue by ianhomer** [vim/vim#9183]
### Steps to reproduce

```shell
echo "{}" > test.jsonc
vim --clean test.jsonc +"filetype plugin on" +"do BufRead" +"do BufRead"
```

causes the following error to be reported

```
Error detected while processing command line..BufRead Autocommands
for "*.jsonc"..FileType Autocommands for "*"..
function <SNR>3_LoadFTPlugin:
line    2:
E492: Not an editor command: commentstring< comments<
```

Note that the following loads without error

```shell
echo "{}" > test.json
vim --clean test.json +"filetype plugin on" +"do BufRead" +"do BufRead"
```

### Expected behaviour

jsonc buffer should be read without error

### Operating system

Linux version 5.10.47 - Dockerfile archlinux:latest 

### Version of Vim

8.2.3582

### Logs and stack traces

```
Error detected while processing command line..BufRead Autocommands for 
"*.jsonc"..FileType Autocommands for "*"..
function <SNR>3_LoadFTPlugin:
line    2:
E492: Not an editor command: commentstring< comments<
```

**Comment by lacygoill** [comment]
Patch fixing the issue:
```diff
diff --git a/runtime/ftplugin/jsonc.vim b/runtime/ftplugin/jsonc.vim
index 90d52cd0d..97ef0498c 100644
--- a/runtime/ftplugin/jsonc.vim
+++ b/runtime/ftplugin/jsonc.vim
@@ -19,7 +19,7 @@ let s:undo_ftplugin = []
 
 " Set comment (formatting) related options. {{{1
 setlocal commentstring=//%s comments=sO:*\ -,mO:*\ \ ,exO:*/,s1:/*,mb:*,ex:*/,://
-call add(s:undo_ftplugin, 'commentstring< comments<')
+call add(s:undo_ftplugin, 'setlocal commentstring< comments<')
 
 " Let Vim know how to disable the plug-in.
 call map(s:undo_ftplugin, "'execute ' . string(v:val)")
```
Although, the code could be simplified a lot, and avoid adding an extra `:execute` (on top of [the default one](https://github.com/vim/vim/blob/88a4205f1cfbdc328e987ab00521fc8a22447fc3/runtime/ftplugin.vim#L16)):
```diff
diff --git a/runtime/ftplugin/jsonc.vim b/runtime/ftplugin/jsonc.vim
index 90d52cd0d..bdbbf30f4 100644
--- a/runtime/ftplugin/jsonc.vim
+++ b/runtime/ftplugin/jsonc.vim
@@ -14,14 +14,8 @@ else
   let b:did_ftplugin_jsonc = 1
 endif
 
-" A list of commands that undo buffer local changes made below.
-let s:undo_ftplugin = []
-
 " Set comment (formatting) related options. {{{1
 setlocal commentstring=//%s comments=sO:*\ -,mO:*\ \ ,exO:*/,s1:/*,mb:*,ex:*/,://
-call add(s:undo_ftplugin, 'commentstring< comments<')
 
 " Let Vim know how to disable the plug-in.
-call map(s:undo_ftplugin, "'execute ' . string(v:val)")
-let b:undo_ftplugin = join(s:undo_ftplugin, ' | ')
-unlet s:undo_ftplugin
+let b:undo_ftplugin = 'setlocal commentstring< comments<'
```

**Comment by brammool** [comment]
The original ftplugin also has the simplified version.
@izhakjakov

**Comment by chrisbra** [comment]
fixed by 6304be625ce44dcfedc6735164d0b853578581c8

**Comment by IzhakJakov** [comment]
Sorry I missed this.  Thank you for the fix :)
